modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/checkpoint/CheckpointHistory.java

```diff
@@ -181,9 +193,102 @@ public WALPointer firstCheckpointPointer() {
      * @param entry Entry to add.
      */
     public void addCheckpoint(CheckpointEntry entry) {
+        addCpToEarliestCpMap(entry);
+
         histMap.put(entry.timestamp(), entry);
     }
 
+    /**
+     * Update map which stored the earliest checkpoint each partitions for groups.
```

Comment:
Updates map which stores the earliest checkpoint for each partitions in every group.

Reply:
Done.